src/oic/oic/__init__.py

```diff
@@ -1064,12 +1066,16 @@ def fetch_distributed_claims(self, userinfo, callback=None):
             if "endpoint" in spec:
                 if "access_token" in spec:
                     _uinfo = self.do_user_info_request(
-                        token=spec["access_token"],
+                        method='GET', token=spec["access_token"],
                         userinfo_endpoint=spec["endpoint"])
                 else:
-                    _uinfo = self.do_user_info_request(
-                        token=callback(csrc),
-                        userinfo_endpoint=spec["endpoint"])
+                    if callback:
```

Comment:
This part is not covered by tests.

Reply:
Does the recent changes meet your requirements ?

Reply:
Sorry for not responding earlier. I am having a busy week at work. I will have a look today.

Reply:
Here again, comparison to `None` and test without callback is missing.

Reply:
OK now ? Need to get this out the door soon.